vscode_extension/package.json

```
@@ -45,6 +45,31 @@
                     "default": "",
                     "pattern": "(\\\\?([^\\/]*[\\/])*)([^\\/]+)$",
                     "description": "The filesystem path to the aac CLI tool."
+                },
+                "aac.lsp.serverMode": {
+                    "scope": "machine-overridable",
+                    "type": "string",
+                    "enum": [
+                        "IO",
+                        "TCP"
+                    ],
+                    "default": "IO",
+                    "description": "The method of communication with the LSP server."
+                },
+                "aac.lsp.tcp.host": {
```

I'd expect host/port to be available for websockets too. Is there a reason to create per-protocol host/port config items or can these be generic amongst the protocols since any networking protocol will need host/port.
```suggestion
                "aac.lsp.host": {
```

---

I thought about this, too. Honestly, I was torn on whether it would matter that much but the main use case I thought of for having two different `host`/`port` configurations (one for TCP and one for WebSockets) was in case someone wanted to switch between LSP server implementations. In that case (with separate `host`/`port` configurations), they could just change the `serverMode` property to the style of server they want to use. Otherwise, they'd have a slightly more error-prone process of setting the `serverMode`, the `host`, and the `port` anytime they wanted to switch (this assumes they would not use the same `host`/`port` for both). It may be a small gain, but from a user's standpoint, I wouldn't want to necessarily have to update it, if I didn't have to.

Again, I doubt it would be a huge deal, I just thought it could offer a bit more ease of use, IMO.